app/assets/src/components/views/bulk_download/BulkDownloadModal.jsx

```diff
@@ -1,62 +1,128 @@
 import React from "react";
 import PropTypes from "prop-types";
-import { set } from "lodash/fp";
+import { find, get, set } from "lodash/fp";
+import memoize from "memoize-one";
 
-import { getBulkDownloadTypes } from "~/api";
+import { getBulkDownloadTypeMetadata } from "~/api";
 import Modal from "~ui/containers/Modal";
 
 import ChooseStep from "./ChooseStep";
+import ReviewStep from "./ReviewStep";
+
+const assembleSelectedDownload = memoize(
```

Comment:
Why do you need to memoize this?
Not sure if I understand whats happening but wouldnt storing the selected on a state variable a simpler alternative?

Reply:
This is IMO a cleaner way to do it.

If we store the selected download in the state, then we would need to remember to update the state value every time certain props changed. Here, the `memoize` only recalculates the value if any of the input parameters changed. So it takes care of all that logic for us.

Granted, the "assembleSelectedDownload" function isn't very expensive, so the memoize is more of a best practice here. Even if we ran it every render, it probably wouldn't be a problem.

Reply:
Ok, I did not know that this library just memoizes the last one (hence the name... ) so seems fine. I actually find it more confusing as you got to know that. But leave it up to you...